apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/RetryTimings.cs

```diff
@@ -0,0 +1,124 @@
+    /// The default initial backoff time between the first attempt and the first retry. It is set to 1 second by default.
+    /// </summary>
+    public static TimeSpan DefaultInitialBackoff { get; } = TimeSpan.FromSeconds(1);
+
+    /// <summary>
+    /// The default maximum backoff time between retries. It is set to 32 seconds by default.
+    /// </summary>
+    public static TimeSpan DefaultMaxBackoff { get; } = TimeSpan.FromSeconds(32);
+
+    /// <summary>
+    /// The default maximum backoff multiplier to be applied on each iteration. It is set to 2 by default.
+    /// </summary>
+    public static double DefaultBackoffMultiplier { get; } = 2;
+
+    /// <summary>
+    /// Returns the retry timing configurations all set to default.
+    /// </summary>
+    public static RetryTimings Default { get; } = new();
+
+    internal RetryTimings Clone() => new()
+    {
+        InitialBackoff = InitialBackoff,
+        MaxBackoff = MaxBackoff,
+        BackoffMultiplier = BackoffMultiplier
+    };
+
+    /// <summary>
+    /// Returns the existing retry timings configurations using the specified initial backoff.
+    /// </summary>
+    /// <param name="initialBackoff">Custom initial Backoff for retry.</param>
+    /// <returns>Retry timings with updated initial backoff.</returns>
+    public RetryTimings WithInitialBackoff(TimeSpan initialBackoff)
+    {
+        var clone = Clone();
+        clone.InitialBackoff = initialBackoff;
+        return clone;
+    }
+
+    /// <summary>
+    /// Returns the existing retry timings configurations using the specified maximum backoff.
+    /// </summary>
+    /// <param name="maxBackoff">Custom maximum Backoff for retry.</param>
+    /// <returns>Retry timings with updated maximum backoff timing.</returns>
+    public RetryTimings WithMaxBackoff(TimeSpan maxBackoff)
+    {
+        var clone = Clone();
+        clone.MaxBackoff = maxBackoff;
+        return clone;
+    }
+
+    /// <summary>
+    /// Returns the existing retry timings configurations using the specified backoff multiplier.
+    /// </summary>
+    /// <param name="backoffMultiplier">Custom retry delay Backoff Multiplier for retry.</param>
+    /// <returns>Retry timings with updated backoff multiplier.</returns>
+    public RetryTimings WithBackoffMultiplier(double backoffMultiplier)
+    {
+        var clone = Clone();
+        clone.BackoffMultiplier = backoffMultiplier;
+        return clone;
+    }
+
+    /// <summary>
+    /// Retry Timings constructor which sets all values to their respective defaults.
+    /// </summary>
+    public RetryTimings()
+    {
+        InitialBackoff = RetryTimings.DefaultInitialBackoff;
+        MaxBackoff = RetryTimings.DefaultMaxBackoff;
+        BackoffMultiplier = RetryTimings.DefaultBackoffMultiplier;
+    }
+
+    /// <summary>
+    /// Constructor to pass the custom timing configurations for retrying.
+    /// </summary>
+    /// <param name="initialBackoff">Initial Backoff for retry.</param>
```

Comment:
Nit: Backoff => backoff in all comments

This is where we should specify the constraints in terms of "must" using `paramRef`.